library/ssl_tls.c

```diff
@@ -4355,7 +4355,9 @@ static int ssl_conf_set_psk_identity( mbedtls_ssl_config *conf,
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
 
-    conf->psk_identity = mbedtls_calloc( 1, psk_identity_len );
+    /* The PSK Identity can be length zero, but we can't allocate zero bytes,
+       so increment, so the buffer is always greater than zero. */
+    conf->psk_identity = mbedtls_calloc( 1, ( psk_identity_len + 1 ) );
     if( conf->psk_identity == NULL )
```

Comment:
```suggestion
    conf->psk_identity = mbedtls_calloc( 1, psk_identity_len );
    if( psk_identity_len != 0 && conf->psk_identity == NULL )
```

Reply:
Slightly embarrassed I didn't see the more obvious, simpler and elegant solution you proposed as opposed to my more complex fix - but anyway - I've added that fix.